### ISSUE TITLE
As a developer, I wish typography groups and variants followed the same naming convention as EDS

### ISSUE THREAD

**Issue by tormodAase** [equinor/mad#125]
When looking at design from figma, it's easy to see which group and variant I should use from `eds-core-react` when developing a webapp:
![image](https://github.com/equinor/mad/assets/22524448/8dba3eda-96e9-42a9-8243-757a6677ad6f)

However, when using `mad-components` it's really difficult.

Here's all the different groups and variants created by EDS (found [here](https://storybook.eds.equinor.com/?path=/docs/typography-typography--docs#overview))
![image](https://github.com/equinor/mad/assets/22524448/20957817-d6df-4296-b031-70f2574b8579)

vs the current groups/variants in mad-components:
![image](https://github.com/equinor/mad/assets/22524448/74e8a080-2385-4872-b85a-76f5add32aef)

**Comment by HaakonSvane** [comment]
I completely agree. This perhaps got a bit rushed when implementing it initially. 
Do you have any ideas on the api? Should we stick to the same groups and variant type of props or should we look into better ways pf doing it?

One idea is to have a Typography component that contains "basic" variants and also have a Typography.Ui, Typography.Table etc. that have scoped variant props (e.g Typography.Table only have cell_header, cell_text etc. as allowed variants)

**Comment by HaakonSvane** [comment]
I subscribed to this idea as well at the beginning of this project, but after seeing outdated practices and areas of the codebase that would benefit from improvement/modifications in eds-core-react, I am not too sure if I have that philosophy anymore.

A good example of where our libraries differ is through the use of child components vs props as a way to define behaviour.

I'm on mobile, so code comments wont work, but look at our implementation of the Accordion va theirs. They rely a lot on child components such as <Accordion.Header> and <Accordion.Content> to structure the accordion. They make an implicit assumtion that for each <Accordion.Item>, there should obly be one corresponding <Accordion.Header> and one <Accordion.Content> inside. There is no way of enforcing this in typescript without going against the framework. I think this is bad design.

**Comment by tormodAase** [comment]
Yeah if there are good reasons to change the API, then I agree we should do so. But I don't see how 
`<Typography.Table variant='cell_header' />` 
is superior to 
`<Typography group="table" variant="cell_header" />`

But if we for some reason want to change the API, we might as well go all in and do 
`<Typography.Table.CellHeader />`

**Comment by tormodAase** [comment]
But I think there is some value to keeping `Typography` as similar to `eds-core-react`'s version as possible, since it's such a frequently used component. It can be annoying to switch from web projects to mobile projects when you have to remember both syntaxes

**Comment by HaakonSvane** [comment]
I think it is better since it communicates that there is a hierarchy between the groups and variants. This is not at all communicated to the user as of today (we tried to create some nice TS typings for it, but there is nothing stopping the user from inserting an invalid variant into a group, and there is no clear way for the user to discover this relationship without guessing based on what the EDS docs state)


To highlight why I think this is a bad design choice, eds core react first establishes this constrained relationship on its props and throws errors if you guess them incorrectly. If you want a paragraph short typography, why should the client app crash until the user can figure out the correct group-variant combination?

If we expose a basic <Typography> component that mirrors the "quick styles" that eds core react have (see their implementation), we can communicate the typography style structure more clearly since each <Typography.{Group}> have a distinct set of variants allowed in contrast to the old way of doing things.

From a pragmatic perspective, I don't often see developers switching between the two libraries too frequently, and when they do, their experience level assures me that they are comfortable with the differences between the two. New developers are more fragile though. Our library should focus on being clear and easy to use and discover because of this in my opinion.

**Comment by HaakonSvane** [comment]
@hathibelagal-dev2 
@dansvend 

Any thoughts on this?

**Comment by tormodAase** [comment]
I guess there aren't many webapps in Mad Max, but in DSD mobile and Flex we have multiple webapps, and atleast in DSD mobile's case, we spend 10-50% of our time in our webapps

**Comment by HavardNot** [comment]
I think this is a good point to think about what we want mad and mad-components to be for the business as a whole. 

If it is to be primarily used by the MAD team and not a generic available library to Equinor developers as a whole, we can be more free to improve upon the implementation methods of EDS to better suite modern standards of mobile development and the team's standards. 

However, if this is to be a library that, say, a team that has created a web app using eds-core-react, can use to create a mobile version of the web app, it can be a huge deterrent for these teams to adopt mad-components if it is a totally different framework with little similarity to EDS. 

_This might be something we have already decided upon already, I dont know._
